OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/alias/AllocationSiteAndTacBasedAliasAnalysis.scala

```diff
@@ -0,0 +1,68 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj.tac.fpcf.analyses.alias
+
+import org.opalj.br.PC
+import org.opalj.br.fpcf.properties.Context
+import org.opalj.br.fpcf.properties.NoContext
+import org.opalj.tac.Goto
+
+trait AllocationSiteAndTacBasedAliasAnalysis extends AllocationSiteBasedAliasAnalysis with TacBasedAliasAnalysis {
+
+    override protected[this] type AnalysisState <: AllocationSiteBasedAliasAnalysisState with TacBasedAliasAnalysisState
+
+    override protected[this] def checkMustAlias(intersection: AllocationSiteBasedAliasSet)(implicit
+        state:   AnalysisState,
+        context: AnalysisContext
+    ): Boolean = {
+
+        val pointsTo1 = state.pointsTo1
+        val pointsTo2 = state.pointsTo2
+
+        if (intersection.size != 1 ||
+            pointsTo1.size != 1 ||
+            pointsTo2.size != 1
+        ) return false
+
+        // they refer to the same allocation site but aren't necessarily the same object (e.g. if the allocation site
+        // is inside a loop or a different method and is executed multiple times)
+
+        val (pointsToContext: Context, pc: PC) = pointsTo1.allPointsTo.head
```

Comment:
You used `.head` here several times. This might be correct, but most often it isn't because it ignores the other elements.

Reply:
Before this line is executed we require that both pointsTo sets contain exactly one element. If that is not the case we directly return false. So that assumption shouldn't be a problem

Reply:
Is `pointsTo1.size` the same as `pointsTo1.allPointsTo.size`? Then yes

Reply:
Yes, it's the same